web/source/keymanweb.js

```diff
@@ -2770,14 +2770,71 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
       else if (e.which) s.Lcode = e.which;    
       else return null;
       
-      s.Lmodifiers = 
-        (e.shiftKey ? 0x10 : 0) |
-        (e.ctrlKey ? (e.ctrlLeft ? 0x20 : 0x20) : 0) | 
-        (e.altKey ? (e.altLeft ? 0x40 : 0x40) : 0);  // I3363 (Build 301)
+      var osk = keymanweb["osk"];
+
+      if(keymanweb.isChiral()) {
+        s.Lmodifiers = (e.shiftKey ? 0x10 : 0);
+
+        var ctrlEvent = false, altEvent = false;
+
+        switch(s.Lcode) {
+          case osk.keyCodes["K_CONTROL"]:
+          case osk.keyCodes["K_CTRL"]:
+          case osk.keyCodes["K_LCONTROL"]:
+          case osk.keyCodes["K_LCTRL"]:
+          case osk.keyCodes["K_RCONTROL"]:
+          case osk.keyCodes["K_RCTRL"]:
+            ctrlEvent = true;
+            break;
+          case osk.keyCodes["K_ALT"]:
+          case osk.keyCodes["K_LALT"]:
+          case osk.keyCodes["K_LMENU"]:
+          case osk.keyCodes["K_RALT"]:
+          case osk.keyCodes["K_RMENU"]:
+            altEvent = true;
+            break;
+        }
+
+        // TODO:  Ensure the keypress is also the CTRL or the ALT for the e.location != 0 check.
+        if(e.ctrlKey) {
+          s.Lmodifiers = s.Lmodifiers | ((e.location != 0 && ctrlEvent) ? 
```

Comment:
I find these two lines very hard to grok

Reply:
There are two separate conditions that should trigger the CTRL modifier.

1) The user literally just pressed the CTRL key, so the event has a valid `location` property we can utilize, but its layer isn't presently activated within the OSK.
2) CTRL has been held a while, so the OSK layer is valid, but the key event doesn't tell us the chirality of the active CTRL press.

In either case, `e.ctrlKey` is set to true, telling us that there is an active CTRL modifier - but not if it's new or maintained from a prior `_KeyDown`.

The `e.location != 0` test passing matches condition 1 above, while its failure matches condition two - hence the layered assignment line.

Reply:
I guess I'm asking if we can rewrite or expand the conditions to make it easier for other codes to grok as well. The documentation that you've just written here would make an excellent code comment.